core/src/main/java/com/graphhopper/routing/template/RoundTripRoutingTemplate.java

```diff
@@ -100,6 +100,14 @@ public RoundTripRoutingTemplate(GHRequest request, GHResponse ghRsp, LocationInd
         return queryResults;
     }
 
+    private double getInitialHeading() {
+        if (ghRequest.hasFavoredHeading(0)) {
+            return ghRequest.getFavoredHeading(0);
+        } else {
+            return Double.NaN;
+        }
+    }
```

Comment:
Do we need this here? Shouldn't ghRequest.getFavoredHeading(0) return NaN already in those cases?

Reply:
Without this check we get `IndexOutOfBoundsException` in case no heading was used because then the favoredHeadings list is empty. Of course it would be possible to change `GHRequest#getFavoredHeading` and make a similar check there, it is otherwise only used in `ViaRoutingTemplate`.

Reply:
ah, I see there is a first condition against the size in the hasFavoredHeading method ... maybe we remove this first check and always fill the `favoredHeadings` List? But to be honest: I do not see how this list stays empty :)

Reply:
Oh ok you are right in GHRequest it is made sure that `favoredHeadings always has same size as `points`, and `favoredHeadings` is initialized with NaN entries.